packages/jupyter-ai/jupyter_ai/document_loaders/directory.py

```diff
@@ -5,13 +5,28 @@
 from typing import List
 
 import dask
+from langchain.document_loaders import PyPDFLoader
 from langchain.schema import Document
 from langchain.text_splitter import TextSplitter
+from pypdf import PdfReader
+
+
+# Uses pypdf which is used by PyPDFLoader from langchain
+def pdf_to_text(path):
+    reader = PdfReader(path)
+    pages = reader.pages
+    text = ""
+    for page in pages:
+        text = text + "\n \n" + page.extract_text()
```

Comment:
It might be quicker to do this using a combination of [map](https://docs.python.org/3/library/functions.html#map) and [str.join](https://docs.python.org/3/library/stdtypes.html#str.join).

Reply:
Code change made, streamlined it down to one line (thanks for this suggestion):
![Streamlines PDF loader for all pages](https://github.com/jupyterlab/jupyter-ai/assets/29005/b10ed654-af02-47f7-8eb1-cbfa0aec50c4)